### ISSUE TITLE
About Loss

### ISSUE THREAD

**Issue by CVUsers** [khdlr/HED-UNet#10]
Hi, I used to train with a loss of 0 x. The effect is good. In the past two days, 1 / 5 of the new data types have been added on the basis of the original data: 6000 white masks (but I want him to learn this type of data because the pixels on the outermost edge are black). At present, 10 epochs have been trained, and loss has increased from 4 to 9, but the edge accuracy has increased slightly, 0.978; The final effect is to successfully learn this new type of data, but it can be more perfect. What causes the loss

**Comment by khdlr** [comment]
That is odd, my assumption would be that the auto-weighting loss is a bit out of control. Here's a possible fix. Try replacing `/deep_learning/loss_functions` with this adapted file, which includes a hybrid loss: https://gist.github.com/khdlr/c0bb6a653ac00c6d6859e6dc773daf96

Then select the hybrid loss function by changing lines 10-11 in `config.yml` to read
```yml
loss_args:
  type: HybridLoss
```

Of course there's no way of knowing whether it will actually help, but it might just do the trick :)

**Comment by CVUsers** [comment]
Thanks.
I want to design or use a loss function similar to box loss and giou loss in target detection. Because the object edge detection of my task is always closed. At present, some gaps and edges will lead to false edge detection. After using the above losses, it may be alleviated.

**Comment by CVUsers** [comment]
I want to know whether this job can train negative samples close to positive samples (labels may be set to be all empty). For example, I train unet-hed to identify the edge of white paper, but some white rectangular objects will also be mistakenly detected as white paper. Is this a common problem of segmenting networks? Or can it be fitted by negative sample training? What to do.

**Comment by khdlr** [comment]
Yeah, this should be easy to fix by adding negative samples to the training set like you suggested.

**Comment by CVUsers** [comment]
How to mark negative samples? Don't you mark it? All black?


---Original---
From: "Konrad ***@***.***&gt;
Date: Thu, Apr 28, 2022 17:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [khdlr/HED-UNet] About Loss (Issue #10)




 
Yeah, this should be easy to fix by adding negative samples to the training set like you suggested.
